client/stores/NotificationStore.ts

```diff
@@ -65,10 +71,7 @@ export const useNotificationStore = create<NotificationStore>((set) => ({
       );
 
       return {
-        notifications: [newNotification, ...state.notifications].slice(
-          0,
-          MAX_NOTIFICATIONS,
-        ),
+        notifications: [newNotification, ...state.notifications],
```

Comment:
I see the user has the ability to remove notifications in the UI, but it's not clear to me what happens to the timeout if the user clicks the `X`? Maybe it’s a good idea to clear all the timeouts in the clearAllNotifications function or is this handled somewhere else?

Reply:
Thanks for pointing out the clearTimeOut! This really helped pinpoint the memory leak bug. I am going to make some changes to the notification store.

Reply:
👍